<launch>
  <node pkg="face_detector_mono" type="run_gaze_follower.sh" name="gaze_follower" output="screen" respawn="false" />
  
  <node pkg="yarp_to_ros_image" type="yarp_to_ros_image" name="yarp_to_ros_image" output="screen" respawn="false" >
    <param  name="output_image_topic" type="string" value="yarp_to_ros_image"/>
    <param  name="port_name" type="string" value="/yarp_to_ros_image"/>
    <param  name="display" value="false"/>
    </node>

  <node pkg="yarp_to_ros_image" type="connect_tool.sh" name="connect1" output="screen" respawn="false" args="/icub/cam/right /yarp_to_ros_image"/>

  <node pkg="face_detector_mono" type="facedetect" name="facedetect" output="screen" respawn="true" >
    <param  name="cascade_name" type="string" value="$(find opencv2)/opencv/share/opencv/haarcascades/haarcascade_frontalface_alt.xml" />
    <param  name="nested_cascade_name" type="string" value="$(find opencv2)/opencv/share/opencv/haarcascades/haarcascade_eye_tree_eyeglasses.xml" />
    <param name="scale" type="double" value="1.3" />
    <param  name="input_image_topic" type="string" value="/yarp_to_ros_image/yarp_to_ros_image"/>
    <param  name="display" value="true"/>
    <param  name="yarp_image" value="true"/>
    </node>
 
 <node pkg="yarp_to_ros_image" type="connect_tool.sh" name="connect2" output="screen" respawn="false" args=" /icub_look_at_head /gaze_tracker/cam_in"/>
</launch>
